src/Paket.Core/Versioning/SemVer.fs

```diff
@@ -47,42 +53,72 @@ type PreRelease =
     { Origin : string
       Name : string
       Values : PreReleaseSegment list }
-    
+      
     static member TryParse (str : string) = 
-        if String.IsNullOrEmpty str then None 
+        if String.IsNullOrEmpty str then None
         else
-            let name = Regex("^(?<name>[a-zA-Z]+)").Match(str).Value
-
+            let getName fromList =
+                match fromList with
+                | AlphaNumeric(a)::_ -> a
+                | _::AlphaNumeric(a)::_ -> a // fallback to 2nd
+                | _ -> ""
+                
             let parse segment =
                 match bigint.TryParse segment with
-                | true, bint -> Numeric bint
-                | false, _ -> AlphaNumeric segment
+                | true, number when number >= 0I -> Numeric number
+                | _ -> AlphaNumeric segment
+                
+            let notEmpty = StringSplitOptions.RemoveEmptyEntries
+            let name, values = 
+                match str.Split([|'.'|],notEmpty) with
+                | [|one|] -> 
+                    let list = one.Split([|'-'|],notEmpty) |> Array.map parse |> List.ofArray
+                    
+                    // semver1-like embedded / inlined prerelease numbers
+                    let name =
+                        match Regex("^(?<name>[a-zA-Z]+)").Match(one) with 
+                        | ex when ex.Success -> ex.Value
+                        | _ -> getName list
+                        
+                    name, list
+                                    
+                | multiple -> //semver2: dashes are ok, inline numbers not
+                
+                    let list = multiple |> Array.map parse |> List.ofArray
+                    getName list, list
                     
-            let values = str.Split([|'.'|]) |> Array.map parse |> List.ofArray
-            Some { Origin = str; Name = name; Values = values}
+            Some { Origin = str; Name = name; Values = values }
+
+    member x.Equals(y) = x.Origin = y.Origin
 
     override x.Equals(yobj) = 
         match yobj with
-        | :? PreRelease as y -> x.Origin = y.Origin
+        | :? PreRelease as y -> x.Equals(y)
         | _ -> false
-
+        
     override x.ToString() = x.Origin
     
     override x.GetHashCode() = hash x.Origin
+    
+    member x.CompareTo(yobj) = 
```

Comment:
changed to tail-recursive list comparison as zipOpt was slower, not debuggable, and keeping semver2 compliance was nontransparent

Reply:
Nice change, can we remove `zipOpt` now? I should never have written that thing.